e2e/trust/sign_test.go

```diff
@@ -0,0 +1,80 @@
+package trust
+
+import (
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"testing"
+)
+
+const notaryURL = "https://notary-server:4443"
+const registryPrefix = "registry:5000"
+
+const alpineImage = "registry:5000/alpine:3.6"
+const busyboxImage = "registry:5000/busybox:1.27.2"
+const localImage = "registry:5000/signlocal:v1"
+const signImage = "registry:5000/sign:v1"
+
+func TestSignWithTrustSign(t *testing.T) {
+	setupConfigFile(t)
+	defer removeConfigFile(t)
+	icmd.RunCmd(icmd.Command("docker", "pull", alpineImage)).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", alpineImage, signImage).Assert(t, icmd.Success)
+	result := icmd.RunCmd(
+		icmd.Command("docker", "-D", "trust", "sign", signImage),
+		withTrustAndPassphrase("root_password", "repo_password"))
+	result.Assert(t, icmd.Success)
+}
+
+func TestSignWithTrustSignlocal(t *testing.T) {
+	setupConfigFile(t)
+	defer removeConfigFile(t)
+	icmd.RunCmd(icmd.Command("docker", "pull", alpineImage)).Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", alpineImage, localImage).Assert(t, icmd.Success)
+	result := icmd.RunCmd(
+		icmd.Command("docker", "-D", "trust", "sign", localImage),
+		withTrustAndPassphrase("root_password", "repo_password"))
+	result.Assert(t, icmd.Success)
+	icmd.RunCommand("docker", "tag", busyboxImage, localImage).Assert(t, icmd.Success)
```

Comment:
If everything above this is "setup" code then it would be great to extract it into a function that is named accordingly (`setupLocallyTrustedImage()` maybe?).  The idea here is to make it really easy for someone reading the test to be able to tell what is setup, and what is the actual command being tested.